synapse/storage/prepare_database.py

```diff
@@ -489,11 +489,10 @@ def _upgrade_existing_database(
                 (v, relative_path),
             )
 
-            cur.execute("DELETE FROM schema_version")
-            cur.execute(
-                "INSERT INTO schema_version (version, upgraded) VALUES (?,?)",
-                (v, True),
-            )
+        cur.execute("DELETE FROM schema_version")
+        cur.execute(
+            "INSERT INTO schema_version (version, upgraded) VALUES (?,?)", (v, True),
+        )
```

Comment:
I wonder if this should happen *before* we run the deltas? Otherwise if we crash out halfway through the deltas we end up in the situation where the database reports `schema_version=58` but we've run half the deltas from schema version 59?

Reply:
Yeah, good point